File: Selenium/Project/Activity7.py
from selenium.webdriver.common.by import By
# Set up the Firefox Driver with WebDriverManager
from selenium.webdriver.firefox.webdriver import WebDriver
from selenium.webdriver.firefox.service import Service as FirefoxService
from webdriver_manager.firefox import GeckoDriverManager

service = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with WebDriver(service=service) as driver:
    # Navigate to the URL
    driver.get("https://alchemy.hguy.co/lms")

    # Print the title of the page
    title = driver.title
    print("Page title is:", title)
    assert title == "Alchemy LMS – An LMS Application"
    driver.find_element(By.XPATH, "//a[text()='My Account']").click()
    myAccountPageTitle = driver.title
    print("My Account Page Title is ", myAccountPageTitle)
    assert myAccountPageTitle == "My Account – Alchemy LMS"

    #All Course page
    courseList = driver.find_elements(By.XPATH, "//div[@class='ld_course_grid col-sm-8 col-md-4 ']")
    courseListCount = len(courseList)
    print("Course List Count = ", courseListCount)

File: Selenium/Project/Activity7.java
package LMS_Project_Activities;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.testng.Assert;
import org.testng.Reporter;
import org.testng.annotations.AfterClass;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.Test;

import java.awt.*;
import java.util.List;

public class Activity7 {
    WebDriver driver;
    @BeforeClass
    public void setUp(){
        WebDriverManager.firefoxdriver().setup();
        driver = new FirefoxDriver();
        driver.get("https://alchemy.hguy.co/lms");
        Reporter.log("Homepage loaded");


    }

    @Test
    public void testMethod1(){
        Assert.assertEquals(driver.getTitle(), "Alchemy LMS – An LMS Application");

        driver.findElement(By.xpath("//a[text()='All Courses']")).click();

        Assert.assertEquals(driver.getTitle(), "All Courses – Alchemy LMS");
        Reporter.log("Newpage loaded");
    }
    @Test
    public void testMethod2(){
        List<WebElement> courseList = driver.findElements(By.xpath("//div[@class='ld_course_grid col-sm-8 col-md-4 ']"));
        System.out.println("Course List Count = " + courseList.size());
    }


    @AfterClass
    public void tearDown(){
        Reporter.log("browser closed");
        driver.quit();
    }
}




